lib/rsa.js

```diff
@@ -887,6 +889,28 @@ pki.rsa.generateKeyPair = function(bits, e, options, callback) {
   // key generation code if available and if parameters are acceptable
   if(!forge.options.usePureJavaScript && callback &&
     bits >= 256 && bits <= 16384 && (e === 0x10001 || e === 3)) {
+    if(_detectNodeCrypto('generateKeyPair')) {
+      return _crypto.generateKeyPair('rsa', {
+        modulusLength: bits,
+        publicExponent: e,
+        publicKeyEncoding: {
+          type: 'pkcs1',
```

Comment:
Should this be 'spki'?

Reply:
Seems like `pki.publicKeyFromPem` will take either. I used `pkcs1` because that's what was done with `window.crypto.subtle.generateKey` a few lines below, but I can change to `spki` if you think that makes more sense.

Reply:
@dlongley Do you have an opinion on this?  I'm not sure what the ramifications are one way or the other.